modules/hostap/src/supp_main.c

```diff
@@ -696,6 +696,14 @@ static int hostapd_disable_iface_cb(struct hostapd_iface *hapd_iface)
 	supplicant_send_wifi_mgmt_ap_status(hapd_iface,
 					    NET_EVENT_WIFI_CMD_AP_DISABLE_RESULT,
 					    WIFI_STATUS_AP_SUCCESS);
+	hostapd_config_free(hapd_iface->conf);
+	hapd_iface->conf = hapd_iface->interfaces->config_read_cb(hapd_iface->config_fname);
+	for (j = 0; j < hapd_iface->num_bss; j++) {
+		hapd = hapd_iface->bss[j];
+		hapd->iconf = hapd_iface->conf;
+		hapd->conf = hapd_iface->conf->bss[j];
+		hapd->driver = hapd_iface->conf->driver;
+	}
```

Comment:
looks like a subset of `hostapd_reload_config` should we call that function instead? like `iface->interfaces->reload_config(iface)`?

Reply:
> no, we shouldn't call hapd_iface->interfaces->config_read_cb(hapd_iface->config_fname) in for loop, we have one config.

It will take care of all cases i.e., one or many configs.

> And we have removed hapd_iface->conf, we should use config_read_cb rather than reload_config, it clears more thoroughly

Can you please elaboarate on `thoroughly`?

Looking at `hostapd_reload_config` this also does does `hostap_init` and `hostapd_enable_iface` which is undesirable during iface disable. Also `hostapd_disable_iface` doesn't seem to do this for a case where the callback isn't set. Overall, this looks generic enough, so, may be we should do this in `hostapd` itself and submit a patch upstream? And also please share the steps to reproduce this i.e., the exact wpa_cli commands, I can try that in my local Linux hostapd and see if the problem persists upstream.

Reply:
while:
    1. hostapd enable and start ap
    2. hostapd disable and close ap
end
you can reproduce issue using above step.

Reply:
I have tried to use hostapd_reload_config, it add some config we doesn't need, our config is from hostapd_read2_config, so we can't use hostapd_reload_config.

Reply:
Thanks, let me check this locally to better understand before I approve

Reply:
I have looked at the code and also tried to repro using vanilla hostap, but couldn't, it works fine, please see below debugs. Every disable/enable uses the same list (unless added by the user), it is not appending the same password again and again.

```
sudo ./hostapd -iwlan0 hostapd.conf 
SAE: Setup SAE Password Identifier element
SAE: Derive PT for password 12345678
SAE: Derive PT for password secretpass
SAE: PT setup completed
wlan0: interface state UNINITIALIZED->ENABLED
wlan0: AP-ENABLED 
wlan0: AP-DISABLED 
nl80211: deinit ifname=wlan0 disabled_11b_rates=0
wlan0: interface state ENABLED->DISABLED
SAE: Setup SAE Password Identifier element
SAE: Derive PT for password 12345678
SAE: Derive PT for password secretpass
SAE: PT setup completed
wlan0: interface state DISABLED->ENABLED
wlan0: AP-ENABLED 
SAE: Setup SAE Password Identifier element
SAE: Derive PT for password testing123
SAE: Derive PT for password 12345678
SAE: Derive PT for password secretpass
SAE: PT setup completed
wlan0: AP-DISABLED 
nl80211: deinit ifname=wlan0 disabled_11b_rates=0
wlan0: interface state ENABLED->DISABLED
SAE: Setup SAE Password Identifier element
SAE: Derive PT for password testing123
SAE: Derive PT for password 12345678
SAE: Derive PT for password secretpass
SAE: PT setup completed
wlan0: interface state DISABLED->ENABLED
wlan0: AP-ENABLED 
^Cwlan0: interface state ENABLED->DISABLED
wlan0: AP-DISABLED 
wlan0: CTRL-EVENT-TERMINATING 
nl80211: deinit ifname=wlan0 disabled_11b_rates=0
SAE: Free SAE password entries
SAE: Free SAE password entry testing123
SAE: Free SAE password entry 12345678
SAE: Free SAE password entry secretpass
```

Looks like the problem is specific to Zephyr's implementation, if so, can you please share the exact commands to reproduce the issue?

Reply:
yes. the problem is from Zephyr's implementation, you can reproduce the issue by the command rather hostapd command.

while
wifi ap enable -s gaofeng_wpa3_gcc -c 149 -p 1234567890 -k 3 -w 2
delay 10s
wifi ap disable
delay 10s
endwhile

Reply:
@jukkar , do you have any concern? the problem is from Zephyr's implementation, if you want to reproduce the issue, you can refer to above comment.

Reply:
@jukkar @krish2718 please help review the patch. if you have any other suggestions, please add comment, and I will update it with your comment.

Reply:
nRF7002 doesn't support HOSTAPD_AP and issue is seen only with that option, so, I cannot verify this, but let's go ahead and merge this fix.